Update glutin to 0.19 and winit to 0.18

acheronfail opened the pull request:
Upgrading the `glutin` and `winit` dependencies.

**Q:** Will this automatically publish a new version or should I bump the version too?

---

Also, something to note in [`glutin v0.19`](https://github.com/tomaka/glutin/blob/master/CHANGELOG.md#version-0190-2018-11-09):

> **Breaking:** The entire API for headless contexts has been removed. Please instead use Context::new() when trying to make a context without a visible window. Also removed headless feature.

I've updated the headless tests to just use `Context::new()`.

acheronfail commented:
> I don't think the tests should be removed; there is still support for headless testing.

No worries, I've updated the commit and changed it just to use `Context::new()` instead. 🙂

alexheretic commented:
@acheronfail we also need a [v0.17] branch pr with a `gfx_window_glutin` version bump to setup a release (ie like last time: #2297)

acheronfail commented:
@alexheretic No worries. I've changed the base branch to `v0.17`, and updated the commits accordingly. 👌 

Let me know if there's anything else I need to do.

kvark commented on examples/blend/main.rs:
```diff
@@ -16,7 +16,7 @@
 extern crate gfx_app;
 extern crate image;
-extern crate winit;
+extern crate glutin;
```
why is this changed to `glutin`?

acheronfail replied:
Since `glutin` itself bundles `winit` (and re-exports it) I thought it might be simpler to just use the `glutin` dependency instead.

I've removed `winit` as a direct dependency (except in `src/window/dxgi`, `src/window/vulcan` and `src/window/metal` where it's required). My reasoning was that it's easier to ensure we have the same version of `winit` across the entire repository if we use the `winit` that's re-exported by `glutin`.

If we add the `winit` dependency instead, then we have to ensure that it's the same version as the `winit` exported by `glutin` anyway.

acheronfail replied:
Happy to change all these back to `winit` and re-include it as well - that's an easy change. You can see what it would look like here: https://github.com/gfx-rs/gfx/compare/v0.17...acheronfail:chore/update-dependencies-with-winit

kvark commented:
Glutin should only be used with GL. The core/render/examples should be backend-agnostic, thus using winit for everything. The  only place where glutin is required is the context initialization for GL.

> On Nov 12, 2018, at 17:38, acheronfail <notifications@github.com> wrote:
> 
> @acheronfail commented on this pull request.
> 
> In examples/blend/main.rs:
> 
> > @@ -16,7 +16,7 @@
>  extern crate gfx;
>  extern crate gfx_app;
>  extern crate image;
> -extern crate winit;
> +extern crate glutin;
> Since glutin itself bundles winit (and re-exports it) I thought it might be simpler to just use the glutin dependency instead.
> 
> I've removed winit as a direct dependency (except in src/window/dxgi, src/window/vulcan and src/window/metal where it's required). My reasoning was that it's easier to ensure we have the same version of winit across the entire repository if we use the winit that's re-exported by glutin.
> 
> If we add the winit dependency instead, then we have to ensure that it's the same version as the winit exported by glutin anyway.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

acheronfail commented:
@kvark Okay, thanks for explaining that. 🙂 
I've reverted it to how it was before (only performing the upgrade, and using `winit` everywhere).

alexheretic commented:
lgtm

kvark commented:
Please bump the minor versions of affected crates before we merge

kvark commented:
(note this wasn't required for pre-ll branch because we aren't publishing from it right now. We'll adjust the versions when we do)

acheronfail commented:
Okay, minor version bumped for:

* `gfx_app` from `0.7.0` to `0.8.0`
* `gfx_window_dxgi` from `0.16.0` to `0.17.0`
* `gfx_window_glutin` from `0.26.0` to `0.27.0`
* `gfx_window_metal` from `0.7.0` to `0.8.0`
* `gfx_window_vulkan` from `0.7.0` to `0.8.0`

kvark commented:
bors r+
